frontend/src/pages/course/Course.tsx

```diff
@@ -22,7 +25,31 @@ const Course: FC = () => {
   const isCourseAdmin = useIsCourseAdmin()
   const navigate = useNavigate()
   const location = useLocation();
+  const [confirmLeaveVisible, setConfirmLeaveVisible] = useState<boolean>(false);
 
+  const handleLeaveConfirm = async () => {
+    try {
+      const result = await leaveCourse(course.courseId.toString());
+      if (result.success) {
+        message.success(result.message);
+        setTimeout(() => {
+          window.location.href = AppRoutes.HOME; // Redirect to home page after leaving course
```

Comment:
Ook hier moet de `useNavigate` hook gebruikt worden.

Reply:
Goed dat je erop wijst, in een eerste fase had ik hier alle functionaliteit gestoken. Pas later heb ik dit verplaatst naar eigen, modulaire bestanden voor ease of use, heel dit stuk werd dus niet gebruikt en heb ik dus verwijderd